fsspec/tests/test_async.py

```diff
@@ -145,3 +145,53 @@ async def go():
         assert fsspec.asyn.running_async()
 
     asyncio.run(go())
+
+
+class DummyAsyncFS(fsspec.asyn.AsyncFileSystem):
+    _file_class = fsspec.asyn.AbstractAsyncStreamedFile
+
+    async def _info(self, path, **kwargs):
+        return {"name": "misc/foo.txt", "type": "file", "size": 100}
+
+    async def open_async(
+        self,
+        path,
+        mode="rb",
+        block_size=None,
+        autocommit=True,
+        cache_options=None,
+        **kwargs,
+    ):
+        return DummyAsyncStreamedFile(
+            self,
+            path,
+            mode,
+            block_size,
+            autocommit,
+            cache_options=cache_options,
+            **kwargs,
+        )
+
+
+class DummyAsyncStreamedFile(fsspec.asyn.AbstractAsyncStreamedFile):
+    async def _fetch_range(self, start, end):
+        return "foo-bar"[start:end]
```

Comment:
I suppose if this should emulate a real streaming file, start!=self.loc is an error, right?

Reply:
Yes. that is correct - but we will never have `start != self.loc` as `self.loc` is being used as a `start` from read method.